diff --git a/CHANGELOG.md b/CHANGELOG.md
deleted file mode 100644
index dad8c06aac26cb8cfced31da5f35ac8e95915f15..0000000000000000000000000000000000000000
diff --git a/lib/get-mapping.js b/lib/get-mapping.js
index 187a02ed65104d8e3a058d6bfe7dc8bd7c285369..c73dae08c131bfa1f5c1d67483fbab0fd0a42921 100644
--- a/lib/get-mapping.js
+++ b/lib/get-mapping.js
@@ -36,13 +36,24 @@ function originalEndPositionFor(sourceMap, generatedEnd) {
     // generated file end location. Note however that this position on its
     // own is not useful because it is the position of the _start_ of the range
     // on the original file, and we want the _end_ of the range.
-    const beforeEndMapping = originalPositionTryBoth(
+    let beforeEndMapping = originalPositionTryBoth(
         sourceMap,
         generatedEnd.line,
         generatedEnd.column - 1
     );
     if (beforeEndMapping.source === null) {
-        return null;
+        let line = generatedEnd.line;
+        while (line >= 1 && beforeEndMapping.source === null) {
+            beforeEndMapping = originalPositionTryBoth(
+                sourceMap,
+                line,
+                Number.MAX_SAFE_INTEGER
+            );
+            line--;
+        }
+        if (beforeEndMapping.source === null) {
+            return null;
+        }
     }
 
     // Convert that original position back to a generated one, with a bump
